roles/edpm_extra_mounts/tasks/extra_mounts.yml

```diff
@@ -29,8 +29,8 @@
 - name: Mount extra and default directories
   become: true
   ansible.posix.mount:
-    name: "{{ item.name }}"
-    src: "{{ item.path }}"
+    path: "{{ item.path }}"
+    src: "{{ item.src | default(omit) }}"
```

Comment:
I think you've to set  default(none), `src` is required when state set to `present, mounted or ephemeral.`

Reply:
hm, is src is required, then this module does not support mounting those w/o src (tmpfs, hugetlbfs)

Reply:
Found some example https://github.com/latouchek/ansible-nfv/blob/master/hugepages.yaml#L17

so src=nodev?

Reply:
I think  I mentioned that you have to use none in those cases.
src: "{{ item.src | default(none) }}"

ramishra@fedora:~$ cat playbook.yaml 
- hosts: localhost
  become: true
  tasks:
    - name: Mount extra and default directories
      become: true
      ansible.posix.mount:
        path: '/dev/hugepages1G'
        fstype: hugetlbfs
        opts: 'pagesize=2M'
        state: mounted
ramishra@fedora:~$ ansible-playbook playbook.yaml 
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match
'all'

PLAY [localhost] *******************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************
ok: [localhost]

TASK [Mount extra and default directories] *****************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "state is mounted but all of the following are missing: src"}

PLAY RECAP *************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

ramishra@fedora:~$ cat playbook.yaml 
- hosts: localhost
  become: true
  tasks:
    - name: Mount extra and default directories
      become: true
      ansible.posix.mount:
        path: '/dev/hugepages1G'
        src: none
        fstype: hugetlbfs
        opts: 'pagesize=2M'
        state: mounted

ramishra@fedora:~$ ansible-playbook playbook.yaml 
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match
'all'

PLAY [localhost] *******************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************
ok: [localhost]

TASK [Mount extra and default directories] *****************************************************************************
changed: [localhost]

PLAY RECAP *************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

Reply:
Thanks, done